master/internal/rm/agentrm/provisioner/aws.go

```diff
@@ -158,69 +161,66 @@ func (c *awsCluster) stateFromEC2State(state *ec2.InstanceState) model.InstanceS
 	return model.Unknown
 }
 
-func (c *awsCluster) prestart(ctx *actor.Context) {
-	if c.SpotEnabled {
-		c.attemptToApproximateClockSkew(ctx)
-		c.cleanupLegacySpotInstances(ctx)
+func (c *awsCluster) Prestart() {
```

Comment:
ticket moving prestart into `new...` for the providers? seems pointless/error prone to have out-of-new initialization that is required.